### PR TITLE
Schema Compact

### PR DESCRIPTION
**Pull request by allenc3**
Initial work on Schema Compact. Almost finished with checking non supported types, just need to implement a cycle detection algorithm to prevent improper message nesting for protobufs.

Added some tests and proto files, and in order to check for annotations in SchemaCompact.java, I had to add a "compile" goal to protoc.

### CODE REVIEW

File: google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1alpha2/SchemaCompact.java [hunk 1]
```diff
@@ -96,4 +118,108 @@ public void check(String tableName, Descriptors.Descriptor userSchema)
...
+   * @param allMessageTypes Keeps track of all message types seen, and make sure they don't repeat
+   *     as recursive protos are not supported.
+   * @return True if field type and option is supported
+   * @throws IllegalArgumentException if schema is invalid
+   */
+  private static boolean isSupportedImpl(
+      Descriptors.Descriptor userSchema, HashSet<Descriptors.Descriptor> allMessageTypes)
+      throws IllegalArgumentException {
+
+    List<Descriptors.OneofDescriptor> oneofs = userSchema.getOneofs();
```
Comment (yirutang) [inline google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1alpha2/SchemaCompact.java:175]: Sorry, the doc is wrong. Oneof can be supported by flattening the fields out. Let's just leave it out of the check.

File: google-cloud-bigquerystorage/pom.xml [hunk 2]
```diff
@@ -108,6 +108,12 @@
       <groupId>org.apache.commons</groupId>
       <artifactId>commons-lang3</artifactId>
     </dependency>
+    <dependency>
+      <groupId>com.google.zetasql</groupId>
```
Comment (stephaniewang526) [inline google-cloud-bigquerystorage/pom.xml:112]: is this a credibly sourced artifact? do we have a more well-known alternative to this?

File: synth.metadata [hunk 3]
```diff
@@ -11,7 +11,7 @@
       "git": {
         "name": ".",
         "remote": "https://github.com/googleapis/java-bigquerystorage.git",
-        "sha": "98750d605dd7a6fead678a0b4573288ddd659277"
+        "sha": "f75eaafb3a20c541191686d1540248b67150709a"
```
Comment (stephaniewang526) [inline synth.metadata:14]: these changes should not be rolled in from this feature PR.

File: .kokoro/build.sh [hunk 4]
```diff
@@ -51,9 +51,7 @@ test)
     RETURN_CODE=$?
     ;;
 lint)
-    mvn \
-      -Penable-samples \
-      com.coveo:fmt-maven-plugin:check
+    mvn com.coveo:fmt-maven-plugin:check
```
Comment (stephaniewang526) [inline .kokoro/build.sh:54]: this change should not be part of this PR
